### ISSUE TITLE
QGIS displays a confusing dialog about 'CRS (%n available)' when receiving a 403 upon WMS getcapabilities

### ISSUE THREAD

**Issue by landryb** [qgis/QGIS#58340]
### What is the bug or the crash?

i have a mapproxy service which depending on various conditions sometimes replies with 401 when the user isn't sending credentials, and sometimes a 403 when the said user correctly authenticates but doesnt have rights on the resource.

instead of displaying a proper error message (like 'unauthorized access') qgis displays 
![Capture d’écran_2024-08-08_09-14-36](https://github.com/user-attachments/assets/e5f76680-65f0-4b83-9297-e602af995f8e)
the message comes from https://github.com/qgis/QGIS/blob/master/src/gui/qgsowssourceselect.cpp#L442 or https://github.com/qgis/QGIS/blob/e1bfad138f39cf3c6231755808372b5e32415d92/src/providers/wms/qgswmssourceselect.cpp#L898, but i dont see a dialog being created here.

if receiving a 401, it displays the login prompt for the service, and if the user cancels that dialog, qgis also displays the same message.

### Steps to reproduce the issue

- add https://tiles.rtge.craig.fr/service as a wms service, without credentials
- connect
- cancel the login prompt
- see the confusing dialog about %n crs available

### Versions

3.38.1

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by landryb** [comment]
there are two cases here, and for one i think https://github.com/qgis/QGIS/blob/master/src/providers/wms/qgswmscapabilities.cpp#L2544 doesnt properly handle 403, but i havent been able to figure out where.. the fact that there's an unrelated message display comes from the dialog created in https://github.com/qgis/QGIS/blob/e1bfad138f39cf3c6231755808372b5e32415d92/src/providers/wms/qgswmssourceselect.cpp#L509 but `lastError()` is uninitialized ? or comes from another object ?

**Comment by landryb** [comment]
that's very strange, because here with my self-build 3.38.1 on openbsd i can reproduce both cases (with the localized french message), and originally its been reported by several of our users using qgis on windows.

it should behave the same with anything returning a 403, barring content-type header values... seems the nginx reverse-proxy retuns an html content for 403s.

it seems you rightfully hit https://github.com/qgis/QGIS/blob/master/src/providers/wms/qgswmscapabilities.cpp#L2720 while i dont. windows-specific codepaths maybe ?

**Comment by landryb** [comment]
oh wow... indeed...
```
    <message>
        <location filename="../src/providers/wms/qgswmscapabilities.cpp" line="2715"/>
        <source>Download of capabilities failed:
%1</source>
        <translation>
Système de coordonnées de référence (%n disponible)</translation>
    </message>
```
quite... unexpected ! :))

**Comment by landryb** [comment]
@agiudiceandrea should this be fixed via a PR against the .ts file (can do so) or it has to go through transifex ?

**Comment by agiudiceandrea** [comment]
@landryb, usually it must be fixed on Transifex: the translated string will be updated in the QGIS source from Transifex on each QGIS release.

**Comment by landryb** [comment]
> @landryb, usually it must be fixed on Transifex: the translated string are updated in the QGIS source from Transifex on each QGIS release.

ok, i've asked to join the qgis project on transifex.

**Comment by landryb** [comment]
fr string fixed in transifex for the 3.34, 3.38 & master branches, so from my understanding of the transifex->github workflow the bug will be fixed with the next releases.
